### ISSUE TITLE
F2 rename of typing.NamedTuple doesn't rename all instances of tuple.

### ISSUE THREAD

**Issue by toppk** [microsoft/pyright#9171]
Type: <b>Bug</b>

# Behaviour

cannot (rename)refactor namedtuple with F2.
## Steps to reproduce:

from dataclasses import dataclass
from typing import NamedTuple
@dataclass
class MyData:
    bar: str

class MyTuple(NamedTuple):
    foo: str

class MyTest:
    a = MyData(bar='bar')
    b = MyTuple(foo='foo')
select on foo on the namedtuple, and I'd expect F2 rename to update all MyTuple like it renamed bar for dataclass does.

# Diagnostic data

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

Extension version: 2024.16.0
VS Code version: Code 1.94.0 (d78a74bcdfad14d5d3b1b782f87255d802b57511, 2024-10-02T13:08:12.626Z)
OS version: Linux x64 6.10.11-100.fc39.x86_64
Modes:

-   Python version (& distribution if applicable, e.g. Anaconda): 3.11.10
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Venv
-   Value of the `python.languageServer` setting: Default

<details>
<summary>User Settings</summary>
<p>

```

languageServer: "Pylance"

testing
• pytestArgs: "<placeholder>"
• pytestEnabled: true

```
</p>
</details>

<details>
<summary>Installed Extensions</summary>

|Extension Name|Extension Id|Version|
|---|---|---|
|Darcula Theme|rokoroku.vscode-theme-darcula|1.2.3|
|GitHub Copilot|GitHub.copilot|1.236.0|
|GitHub Copilot Chat|GitHub.copilot-chat|0.21.0|
|JavaScript Debugger|ms-vscode.js-debug|1.94.0|
|JavaScript Debugger Companion Extension|ms-vscode.js-debug-companion|1.1.3|
|Pylance|ms-python.vscode-pylance|2024.10.1|
|Python|ms-python.python|2024.16.0|
|Python Debugger|ms-python.debugpy|2024.10.0|
|Ruff|charliermarsh.ruff|2024.50.0|
|Table Visualizer for JavaScript Profiles|ms-vscode.vscode-js-profile-table|1.0.10|
</details>
<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) CPU E3-1275 v3 @ 3.50GHz (8 x 3515)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off<br>webnn: disabled_off|
|Load (avg)|2, 2, 2|
|Memory (System)|31.28GB (8.30GB free)|
|Process Argv|.|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|gnome|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome|
|XDG_SESSION_TYPE|wayland|
</details>

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.384

**Comment by toppk** [comment]
tested this on pylance pre-release and it works great.  thanks for the quick action!
